### PR TITLE
Feat/dialoge drag

### PR DESCRIPTION
**Pull request by Inbal-Tish**

### CODE REVIEW

File: src/components/dialog/index.js [hunk 1]
```diff
@@ -53,85 +54,145 @@ class Dialog extends BaseComponent {
...
+  }
+
+  onModalShow = () => {
+    const {animationConfig} = this.getThemeProps();
+    const {deltaY} = this.state;
+ 
+    Animated.timing(deltaY, {
+      toValue: 0,
+      duration: _.get(animationConfig, 'duration', 280),
+      delay: _.get(animationConfig, 'delay', 200),
```
Comment (ethanshar) [inline src/components/dialog/index.js]: the default delay feels like a bug

File: src/components/dialog/index.js [hunk 2]
```diff
@@ -53,85 +54,145 @@ class Dialog extends BaseComponent {
...
+    );
+  }
+
+  renderAnimationContainer() {
+    const {animationConfig, top} = this.getThemeProps();
+    const {alignments, deltaY} = this.state;
+    const centerByDefault = _.isEmpty(alignments);
     const animation = top ? AnimatableManager.presets.slideInDown : AnimatableManager.presets.slideInUp;
 
+    if (animationConfig && animationConfig.animation) {
```
Comment (ethanshar) [inline src/components/dialog/index.js]: The two renders are almost identical...
maybe we can do the trick when we if-else just the type of container and then render it.

File: demo/src/screens/componentScreens/DialogScreen.js [hunk 3]
```diff
@@ -91,9 +104,21 @@ class DialogScreen extends Component {
           centerH
           onDismiss={() => this.setState({showDialog5: false})}
           animationConfig={{duration: 1000}}
```
Comment (ethanshar) [inline demo/src/screens/componentScreens/DialogScreen.js]: let's remove the animationConfig here so the example will look better

File: src/components/panGestureView/index.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  };
+  handlePanResponderMove = (e, gestureState) => {
+    const {direction} = this.getThemeProps();
+    let newValue = 0;
+    
+    // VERTICAL
+    const up = (direction === DIRECTIONS.UP);
+    const {deltaY} = this.state;
+
+    if (Math.abs(gestureState.vy) >= 1.8) {
```
Comment (ethanshar) [inline src/components/panGestureView/index.js]: extract the weird values like `gestureState.vy` to constants and give them better naming also they repeat themselves

File: src/components/panGestureView/index.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      if ((up && gestureState.vy < 0) || (!up && gestureState.vy > 0)) {
+        // Swipe
+        this.swipe = true;
+      }
+    } else if ((up && gestureState.dy < 0) || (!up && gestureState.dy > 0)) {
+      // Drag
+      newValue = gestureState.dy;
+      Animated.spring(deltaY, {
```
Comment (ethanshar) [inline src/components/panGestureView/index.js]: `Animated.spring` being called twice almost the same, I would move to a function with a param to control the `toValue`

File: src/components/panGestureView/index.js (hunk 4 above)
Comment (ethanshar) [inline src/components/panGestureView/index.js]: `1.8` is to arbitrary, maybe we can put in a const if it has a meaning

File: demo/src/screens/componentScreens/DialogScreen.js (hunk 3 above)
Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/DialogScreen.js]: done

File: src/components/panGestureView/index.js (hunk 4 above)
Comment (Inbal-Tish) [reply src/components/panGestureView/index.js]: done

Comment (Inbal-Tish) [reply src/components/panGestureView/index.js]: done

**Pull request merged**